packages/mermaid/src/rendering-util/handle-markdown-text.ts

```diff
@@ -2,16 +2,21 @@ import type { Content } from 'mdast';
 import { fromMarkdown } from 'mdast-util-from-markdown';
 import { dedent } from 'ts-dedent';
 import type { MarkdownLine, MarkdownWordType } from './types.js';
+import { getConfig } from '../config.js';
 
 /**
  * @param markdown - markdown to process
  * @returns processed markdown
  */
 function preprocessMarkdown(markdown: string): string {
+  const markdownAutoWrap = getConfig().markdownAutoWrap;
```

Comment:
Please pass the config to the function, instead of importing `getConfig`.
This makes the function independent and will make writing tests easier.

Reply:
Thank you for the feedback!
I understand what you mean. But I want to be sure, that I get the task correct 😄 .

The preprocessMarkdown function is not a funtion I created new, but it is part of the already existing markdown functionality. If I change the parameters of these functions (the same was done in the createTexts function), I will need to change all function calls. If this function (preprocessMarkdown or createText) is called by a function that does not know about config, I will need to change this calling functions definition as well. And so on until I encounter a module that already imports getConfig.

Maybe I am missing a more simple solution here. 
I just wanted to be sure, I understand your requested change before I go and change - protentially - a lot of function definitions.

Reply:
Nope, your suggested approach is right. 
You will have to modify `markdownToLines` and `createText`. 

You'll have to add an optional parameter here

https://github.com/mermaid-js/mermaid/blob/8be8736131cb8068eb9c1b64963ab6f53d38025d/packages/mermaid/src/rendering-util/createText.ts#L156-L168

---

Context: There is a plan to improve the internal architecture, removing global variables and other dependencies that could get polluted after rendering. This would enable us to remove the render queue and have proper async (parallel) rendering. 

So we should avoid new imports of getConfig in internal functions.